mne/decoding/time_gen.py

```diff
@@ -203,16 +192,31 @@ def predict(self, epochs):
         if not hasattr(self, 'estimators_'):
             raise RuntimeError('Please fit models before trying to predict')
 
-        # clean attributes
+        # Check predict mode
+        if self.predict_mode not in ['cross-validation', 'mean-prediction']:
+            raise ValueError('`predict_mode` must be a str, "mean-prediction" '
+                             'or "cross-validation"')
+
+        # Check that training cv and predicting cv match
+        if self.predict_mode == 'cross-validation':
+            n_est_cv = [len(estimator) for estimator in self.estimators_]
+            inconsistent_splits = len(set(n_est_cv)) != 1
+            mismatching_splits = n_est_cv[0] != len(self._cv_splits)
+            mismatching_y = len(self.y_train_) != len(epochs)
```

Comment:
This is recommended python style: http://docs.python-guide.org/en/latest/writing/style/

Having extra temporary variable is not a problem: it doesn't slow down memory.